docs/dev-installation.rst

```diff
@@ -0,0 +1,206 @@
+
+Developer Installation Guide
+============================
+
+If you wish to develop Girder or Girder plugins, you can either
+:ref:`install Girder natively <native_install>` on your machine or
```

Comment:
I'd like to push Vagrant as the default way for new developers to be exposed to Girder:
* It's fewer commands, and platform-agnostic, making it way less effort and less intimidating, which lowers barriers to entry.
* It's far safer for them, as their host machine won't be cluttered with things like MongoDB, etc. that they may not want otherwise.
* It's safer for us, as we don't have to account for the machine's prior state, and get support requests / negative PR from people that were unable to install easily.

So, if @girder/developers agree, can we reorder these docs to place Vagrant first, and perhaps emphasize that this is preferred (including in our trainings and other communications)? Power users are always free to install it on bare metal, but we should let them opt into that set of risk / rewards.

To support this, I hope to port Vagrant to Ubuntu 16.04 soon.